Tools/peg_generator/pegen/c_generator.py

```diff
@@ -682,10 +674,12 @@ def handle_alt_loop(self, node: Alt, is_gather: bool, rulename: Optional[str]) -
             self.print("if (_n == _children_capacity) {")
             with self.indent():
                 self.print("_children_capacity *= 2;")
-                self.print(
-                    "_children = PyMem_Realloc(_children, _children_capacity*sizeof(void *));"
-                )
-                self.out_of_memory_return(f"!_children", "NULL", message=f"realloc {rulename}")
+                self.print("void **_new_children = PyMem_Realloc(_children, _children_capacity*sizeof(void *));")
+                self.out_of_memory_return(f"!_new_children")
+                self.print("else {") # out_of_memory_return generates an if
+                with self.indent():
+                    self.print("_children = _new_children;")
+                self.print("}")
```

Comment:
I think we can eliminate this `else`:
```suggestion
                self.print("_children = _new_children;")
```

Reply:
Yup.